cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -121,24 +122,21 @@ func newEtcdClient() (client *etcd.Client) {
 		}
 	}
 	// loop until we have > 0 machines && machines[0] != ""
-	timeout := false
-	go func() {
-		<-time.After(10 * time.Second)
-		timeout = true
-	}()
-	for !timeout {
-		client = etcd.NewClient([]string{*etcd_server})
-		if client == nil {
-			return nil
+	poll, timeout := 1*time.Second, 10*time.Second
+	if err := wait.Poll(poll, timeout, func() (bool, error) {
+		if client = etcd.NewClient([]string{*etcd_server}); client == nil {
+			log.Fatal("etcd.NewClient returned nil")
 		}
 		client.SyncCluster()
```

Comment:
You're potentially accessing a nil pointer here.  You need to return in the above if statement.